src/main/kotlin/lotto/domain/WinnerStat.kt

```diff
@@ -0,0 +1,31 @@
+package lotto.domain
+
+class WinnerStat(private val purchaseRecord: PurchaseRecord, private val winnerNumbers: List<Int>) {
+    val winnerMap: Map<Winner, Int>
+
+    init {
+        winnerMap = purchaseRecord
+            .lottoList
+            .map { lotto -> getWinner(lotto, winnerNumbers) }
+            .filterNotNull()
+            .groupingBy { it }
+            .eachCount()
+    }
+
+    fun per(): Double {
+        val price = purchaseRecord.lottoList.size * LottoShop.LOTTO_PRICE
+        val earning = winnerMap.map { e -> e.key.prizeMonery * e.value }.sum()
+        return earning / price.toDouble()
+    }
+
+    private fun getWinner(lotto: Lotto, winnerNumbers: List<Int>): Winner? {
+        return Winner.values()
+            .find { it.matchedNumbers == getMatchedNumbers(lotto, winnerNumbers) }
+    }
+
+    private fun getMatchedNumbers(lotto: Lotto, winnerNumbers: List<Int>): Int {
+        return winnerNumbers
+            .filter { lotto.contains(it) }
+            .count()
```

Comment:
```suggestion
            .count { lotto.contains(it) }
```
count도 조건문과 함께 사용할 수 있습니다.